cmd/rook/ceph/cleanup.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ceph
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/cmd/rook/rook"
+	"github.com/rook/rook/pkg/util/flags"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dataDirHostPath string
+	namespaceDir    string
+)
+
+var cleanUpCmd = &cobra.Command{
+	Use:   "clean",
+	Short: "Starts the cleanup process on the disks after ceph cluster is deleted",
+}
+
+func init() {
+	cleanUpCmd.Flags().StringVar(&dataDirHostPath, "data-dir-host-path", "", "dataDirHostPath on the node")
+	cleanUpCmd.Flags().StringVar(&namespaceDir, "namespace-dir", "", "dataDirHostPath on the node")
+	flags.SetFlagsFromEnv(cleanUpCmd.Flags(), rook.RookEnvVarPrefix)
+	cleanUpCmd.RunE = startCleanUp
+}
+
+func startCleanUp(cmd *cobra.Command, args []string) error {
+	rook.SetLogLevel()
+	rook.LogStartupInfo(cleanUpCmd.Flags())
+
+	logger.Info("starting cluster clean up")
+
+	// Delete dataDirHostPath
+	if dataDirHostPath != "" {
+		if err := os.RemoveAll(path.Join(dataDirHostPath, namespaceDir)); err != nil {
+			return errors.Wrap(err, "failed to clean up dataDirHostPath")
```

Comment:
Let's make this a best-effort implementation. In other words, instead of returning when we hit an error, let's log the error and continue to attempt to cleanup everything.

Reply:
How would we know if the job itself failed?

Reply:
As discussed, let's log the error, but then continue attempting to delete other directories and then return success for the job. If the admin wants to investigate why something didn't get cleaned up, they can look at the logs